### PR TITLE
upgrade packages so that zcache is node12-compatible

### PR DESCRIPTION
**Pull request by nicks**
Hello @xiao, @sugunasubra, 

Please review the following commits I made in branch 'nick-snappy'.

9de3adc3637c88d486aa2e278c515bc10e3176e6 (2015-02-12 12:16:53 -0500)
upgrade packages so that zcache is node12-compatible

de87d033b724f3ae8c3f00229a0f6bf411223748 (2015-02-12 12:11:59 -0500)
upgrade snappy. The new version of snappy has a lot of API changes and is async-only.

R=@xiao
R=@sugunasubra

### CODE REVIEW

File: lib/RedisConnection.js [hunk 1]
```diff
@@ -47,42 +47,52 @@ RedisConnection.prototype.isAvailable = function () {
...
-    // Use "MSET" to set all the keys and "EXPIRE" to set TTL for each key
-    var msetCommand = ['MSET']
-    commands.push(msetCommand)
-    for (i = 0, l = items.length; i < l; i++) {
-      var key = items[i].key
-      // Append key value arguments to the set command.
-      msetCommand.push(key, this._compress(items[i].value))
-      // Append an expire command.
-      commands.push(['EXPIRE', key, Math.floor(maxAgeMs / 1000)])
+  var compressedPromises = items.map(function (item) {
```
Comment (xiao) [inline lib/RedisConnection.js]: I'm not sure the performance implication of this approach (compress all and `MSET`) vs. compress and `SET` individual items and `SET` and run them in parallel. With the latter approach, if some item takes long to compress, we would `SET` other shorter items earlier.

**Comment by xiao** [comment]
LGTM. I have a question, not need to address now.

Thanks for the upgrade, nick!

File: lib/RedisConnection.js (hunk 1 above)
Comment (nicks) [inline lib/RedisConnection.js]: note that I haven't changed this behavior. The old code compressed everything up-front too.

Comment (xiao) [inline lib/RedisConnection.js]: It doesn't change what this function does, but it changes the order.

In the old code, we compress everything up-front, but they are all done in one single computation unit.

In the new code, the computation is split into smaller pieces (assuming snappy is smart enough to do that), so we can interleave the call to Redis in the gaps for the items that finish compressing earlier than others. The total amount of time to cache a set of keys might be shorter.

Admittedly, multiple Redis calls will introduce overhead too, so I'm not sure if it's worth it.

**Comment by nicks** [comment]
also bumping the version

**Pull request merged**